components/places/ios/Places/HistoryMetadata.swift

```diff
@@ -28,6 +43,16 @@ public enum HistoryMetadataObservation {
     case titleObservation(String)
     case viewTimeObservation(Int32)
     case documentTypeObservation(DocumentType)
+    case unknown
+    
+    public init(value: Any) {
+        switch value {
+        case let title as String: self = .titleObservation(title)
```

Comment:
What if we have string observations that aren't about `title`, or Int32 that aren't `time`? Also, what's the use case here for the `init`, can't you just do `HistoryMetadataObservation.titleObservation("hello")`?

Reply:
The major issue is that we need some sort of initializer for using it on Firefox iOS side otherwise structs and enums are inaccessible. This is just something how swift hasn't gotten to or a limitation I would say as all this is extra work 😓

Reply:
Oh wow, so that's actually a thing! https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html#//apple_ref/doc/uid/TP40014097-CH41-ID3 (memberwise init section). That's pretty annoying :)

Maybe throw something like `invalidArgumentException` to avoid needing to have an `unknown` case..?